go.mod

```diff
@@ -1,11 +1,16 @@
-module github.com/ddelnano/terraform-provider-xenorchestra
+module github.com/vatesfr/terraform-provider-xenorchestra
 
-go 1.20
+go 1.22
+
+toolchain go1.22.0
```

Comment:
If upgrading the go version is required, it should be separated out into a different change. The github.com/hashicorp/terraform-plugin-sdk project is still on [go 1.21](https://github.com/hashicorp/terraform-plugin-sdk/blob/71cadbe1028da5b6ce844100e42a9d481593f141/go.mod#L3), so I don't think we should upgrade to go 1.22 unless there is a strong need.

Reply:
This also requires Jenkins side changes to upgrade the go version.